toolkit/docs/how_it_works/0_intro.md

```diff
@@ -2,23 +2,308 @@ How The Build System Works
+    rpms --> buildImage
+    buildImage --> image
+
+```
+
+</br>
 
 ### Tools
 The tooling consists of a set of `Makefiles`, various go programs, a bootstrapping environment running in `Docker`, and a `chroot` environment to build in. These are discussed [here](1_initial_prep.md). The toolkit is able to re-build all of the tools from source if desired.
 
+```mermaid
+flowchart TD
+
+    %% style definitions
+    classDef io fill:#247BA0,stroke:#333,stroke-width:2px,color:#fff;
+    classDef process fill:#B05E2F,stroke:#333,stroke-width:2px,color:#fff;
+    classDef decision fill:#51344D,stroke:#333,stroke-width:2px,color:#fff;
+    classDef goodState fill:#566E40,stroke:#333,stroke-width:2px,color:#fff;
+    classDef badState fill:#BC4B51,stroke:#333,stroke-width:2px,color:#fff;
+
+    %% state nodes
+    start(["Start (make toolchain)"]):::goodState
+    done([Done]):::goodState
+    %% error[[Error]]:::badState
+
+    %% TC nodes
+    tcManifests[/Local Toolchain Manifests/]:::io
+    tcRebuild{{"Rebuild Toolchain? (REBUILD_TOOLCHAIN=y/n)"}}:::decision
+    toolchainChoice{{"Toolchain archive available? (TOOLCHAIN_ARCHIVE=...)"}}:::decision
+    tcPopulated([Toolchain populated]):::goodState
+    tcRPMs[(Toolchain RPMs)]:::io
+    tcArchiveOld[/Old Toolchain archive/]:::io
+    tcArchiveNew[/Toolchain archive/]:::io
+    pullTC([Local Toolchain Archive available]):::goodState
+    hydrateTC[Extract Toolchain RPMs]:::process
+    buildRawTC[Build Raw Toolchain]:::process
+    sources[/Sources/]:::io
+    localSpecs[/Local SPECS/]:::io
+    bsRPMS[/Bootstraped Environment/]:::io
+    buildTC[Build Toolchain proper]:::process
+    pullRemote[Download remote RPMs]:::process
+
+    %% TC flow
+    start --> tcRebuild
+    tcRebuild -->|yes| buildRawTC
+    sources-->buildRawTC
+    tcManifests --> hydrateTC
+    tcManifests --> pullRemote
+    tcManifests --> buildTC
+    tcRebuild -->|no|toolchainChoice
+    toolchainChoice -->|no| pullRemote
+    toolchainChoice -->|yes| pullTC 
+    pullRemote --> tcRPMs
+    pullTC -->hydrateTC
+    tcArchiveOld --> pullTC
+    hydrateTC --> tcRPMs
+    buildRawTC --> bsRPMS
+    bsRPMS --> buildTC
+    sources-->buildTC
+    localSpecs --> buildTC
+    buildTC --> tcArchiveNew
+    tcArchiveNew --> pullTC
+    tcRPMs-->tcPopulated
+    tcPopulated --> done
+```
+
+<br/>
+
 ### Package Generation
 Package generation is discussed in detail [here](2_local_packages.md) and [here](3_package_building.md). At a high level, the build system scans the local project directory for `*.spec` files which define the local packages. It then packages these files into `*.src.rpm` files by either including local source files, or downloading matching ones from a source server. The `*.src.rpm` files required for the currently selected configuration are then built in the correct order to resolve all dependencies. The build system will automatically download any build dependencies not satisfied by the local packages.
 
+```mermaid
```

Comment:
- I added "chroot" to the worker step. I would argue this graph is already complicated and it may be worth to keep a little more generalized for now.

- Agreed. I added blocks "Add dependencies to worker" and "Build RPMs" to clean this up. 

- added inputs to "Add dependencies to worker"